### ISSUE TITLE
"selective_scan" is not found

### ISSUE THREAD

**Issue by StudyCrw** [JCruan519/VM-UNet#36]
![image](https://github.com/JCruan519/VM-UNet/assets/141615683/b3b4b544-dd1a-4334-9634-9d99e84ce8da)
I already installed causal_conv1d and mamba_ssm.   However, an error occurred as shown in the figure, how can I slove it? thanks!

**Comment by LHY0329** [comment]
I also have this problem. Have you found a solution?

**Comment by StudyCrw** [comment]
> I also have this problem. Have you found a solution?

I'm sorry, I haven't found a way to solve the problem yet.

**Comment by LHY0329** [comment]
> > I also have this problem. Have you found a solution?
> 
> I'm sorry, I haven't found a way to solve the problem yet.

Okay, thank you!

**Comment by hafsa390** [comment]
I am also facing similar issue. Did you find any solution?

**Comment by StudyCrw** [comment]
> 我
I haven't found a solution either

**Comment by StudyCrw** [comment]
> @LHY0329 , @StudyCrw My cuda version is 12.2. I have installed mamba-ssm=1.0.1 and casual-conv1d=1.2.0post2. I am facing the NameError: name 'selective_scan_fn' is not defined. Could you solve the issue you faced?

I haven't found a solution either.

**Comment by Chang688** [comment]
@LHY0329 您好？请问您解决这个问题了吗？如果解决的话可以告诉我下吗 ？

**Comment by Chang688** [comment]
@StudyCrw 您好？请问您解决这个问题了吗？如果解决的话可以告诉我下吗 ？

**Comment by Chang688** [comment]
@hafsa390 您好？请问您解决这个问题了吗？如果解决的话可以告诉我下吗 ？

**Comment by LHY0329** [comment]
> I haven't found a solution either.

**Comment by LHY0329** [comment]
> @LHY0329 , @StudyCrw My cuda version is 12.2. I have installed mamba-ssm=1.0.1 and casual-conv1d=1.2.0post2. I am facing the NameError: name 'selective_scan_fn' is not defined. Could you solve the issue you faced?

I haven't found a solution either.

**Comment by hafsa390** [comment]
> @hafsa390 您好？请问您解决这个问题了吗？如果解决的话可以告诉我下吗 ？

I have installed the mamba-ssm and causal_conv1d packages using the .whl files from here: https://github.com/state-spaces/mamba/releases and https://github.com/Dao-AILab/causal-conv1d/releases. After that those packages were installed correctly.

**Comment by StudyCrw** [comment]
> > @hafsa390 您好？请问您解决这个问题了吗？如果解决的话可以告诉我下吗 ？
> 
> I have installed the mamba-ssm and causal_conv1d packages using the .whl files from here: https://github.com/state-spaces/mamba/releases and https://github.com/Dao-AILab/causal-conv1d/releases. After that the error did not show up.

Which version do you have installed?

**Comment by onion-LHJ** [comment]
Actually,u just need to make sure ur cuda's version is 11.8,use nvcc -V and check it.If the version is not 11.8,u can use:
 conda install cudatoolkit==11.8 -c nvidia
conda install -c "nvidia/label/cuda-11.8.0" cuda-nvcc

Then use nvcc -V to see the version

Next,u just follow the README,it should work.

**Comment by StudyCrw** [comment]
> Actually,u just need to make sure ur cuda's version is 11.8,use nvcc -V and check it.If the version is not 11.8,u can use: conda install cudatoolkit==11.8 -c nvidia conda install -c "nvidia/label/cuda-11.8.0" cuda-nvcc
> 
> Then use nvcc -V to see the version
> 
> Next,u just follow the README,it should work.

Sorry for the late reply. Thank you very much for your answer.

**Comment by RoryShao** [comment]
Yes, I just solved it. The main issue was due to the CUDA version, which should be higher than 11.8. The problem occurred because the virtual environment was set to 11.7.

**Comment by chjwhere** [comment]
> Actually,u just need to make sure ur cuda's version is 11.8,use nvcc -V and check it.If the version is not 11.8,u can use: conda install cudatoolkit==11.8 -c nvidia conda install -c "nvidia/label/cuda-11.8.0" cuda-nvcc
> 
> Then use nvcc -V to see the version
> 
> Next,u just follow the README,it should work.

Very helpful, thank you:)
